packages/twenty-front/src/modules/object-record/record-field/form-types/components/FormNumberFieldInput.tsx

```diff
@@ -0,0 +1,197 @@
+    multiline ? theme.border.radius.sm : 'none'};
+  border-right: ${({ multiline }) => (multiline ? 'auto' : 'none')};
+  border-top-left-radius: ${({ theme }) => theme.border.radius.sm};
+  border-top-right-radius: ${({ multiline, theme }) =>
+    multiline ? theme.border.radius.sm : 'none'};
+  box-sizing: border-box;
+  display: flex;
+  height: ${({ multiline }) => (multiline ? 'auto' : `${1.5 * LINE_HEIGHT}px`)};
+  overflow: ${({ multiline }) => (multiline ? 'auto' : 'hidden')};
+  /* padding-right: ${({ multiline, theme }) =>
+    multiline ? theme.spacing(6) : theme.spacing(2)}; */
+  width: 100%;
+`;
+
+const StyledInput = styled.input`
+  ${TEXT_INPUT_STYLE}
+
+  padding: ${({ theme }) => `${theme.spacing(1)} ${theme.spacing(2)}`};
+  width: 100%;
+`;
+
+const StyledVariableContainer = styled.div`
+  ${VARIABLE_TAG_STYLES}
+
+  margin: ${({ theme }) => `${theme.spacing(1)} ${theme.spacing(2)}`};
+  align-self: center;
+
+  display: flex;
+  align-items: center;
+`;
+
+const StyledSearchVariablesDropdownContainer = styled.div<{
+  multiline?: boolean;
+  readonly?: boolean;
+}>`
+  align-items: center;
+  display: flex;
+  justify-content: center;
+
+  ${({ theme, readonly }) =>
+    !readonly &&
+    `
+      :hover {
+        background-color: ${theme.background.transparent.light};
+      }`}
+
+  ${({ theme, multiline }) =>
+    multiline
+      ? `
+        position: absolute;
+        top: ${theme.spacing(0)};
+        right: ${theme.spacing(0)};
+        padding: ${theme.spacing(0.5)} ${theme.spacing(0)};
+        border-radius: ${theme.border.radius.sm};
+      `
+      : `
+        background-color: ${theme.background.transparent.lighter};
+        border-top-right-radius: ${theme.border.radius.sm};
+        border-bottom-right-radius: ${theme.border.radius.sm};
+        border: 1px solid ${theme.border.color.medium};
+      `}
+`;
+
+type EditingMode = 'input' | 'variable';
+
+type FormNumberFieldInputProps = {
+  placeholder: string;
+  defaultValue: string | undefined;
+  onPersist: (value: number | null | string) => void;
+};
+
+export const FormNumberFieldInput = ({
+  placeholder,
+  defaultValue,
+  onPersist,
+}: FormNumberFieldInputProps) => {
+  const theme = useTheme();
+
+  const id = useId();
+
+  const [draftValue, setDraftValue] = useState(defaultValue ?? '');
+  const [editingMode, setEditingMode] = useState<EditingMode>(() => {
+    return defaultValue?.startsWith('{{') ? 'variable' : 'input';
+  });
+
+  const persistNumber = (newValue: string) => {
+    if (!canBeCastAsNumberOrNull(newValue)) {
+      return;
+    }
+
+    const castedValue = castAsNumberOrNull(newValue);
+
+    onPersist(castedValue);
+  };
+
+  const handleChange = (newText: string) => {
+    setDraftValue(newText);
+
+    persistNumber(newText.trim());
+  };
+
+  return (
+    <StyledContainer>
+      <StyledInputContainer>
+        <StyledInputContainer2>
+          {editingMode === 'input' ? (
+            <StyledInput
```

Comment:
let's separate:
- FormNumberFieldInput should be a simple number input with a FormOverlay to add design
- Variable button / dropdown should come from the parent